### DISCUSSION TITLE
Do we have maximum-entropy-based chunker?

### DISCUSSION THREAD

**Discussion by wildfluss** [explosion/spaCy#12061] [Help: Coding & Implementations] [answered]
Do we have maximum-entropy-based sentence chunker? Eg, for 

```
Michael McGinn is the mayor of Seattle.
```

it should output

```
["B-NP", "I-NP", "B-VP", "B-NP", "I-NP", "B-PP", "B-NP", "O"]
```

OpenNLP has that here https://opennlp.apache.org/docs/1.7.1/apidocs/opennlp-tools/opennlp/tools/chunker/ChunkerME.html 

I wonder how hard would it be to have the same thing in spaCy.

**Comment by thomashacker** [comment, marked as answer]
Hello,

Currently, spaCy does not have this specific functionality. But it's possible to wrap the OpenNLP chunker to a custom spaCy component and use it in a pipeline. You can find more information about [creating custom components in our docs](https://spacy.io/usage/processing-pipelines/#custom-components).
